dspace-api/src/main/java/org/dspace/app/requestitem/RequestItemServiceImpl.java

```diff
@@ -78,4 +98,28 @@ public void update(Context context, RequestItem requestItem) {
             log.error(e.getMessage());
         }
     }
+
+    @Override
+    public void delete(Context context, RequestItem requestItem) {
+        log.debug(LogHelper.getHeader(context, "delete_itemrequest", "request_id={}"),
+                requestItem.getID());
+        try {
+            requestItemDAO.delete(context, requestItem);
+        } catch (SQLException e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean isRestricted(Context context, DSpaceObject o)
+            throws SQLException {
+        List<ResourcePolicy> policies = authorizeService
+                .getPoliciesActionFilter(context, o, Constants.READ);
+        for (ResourcePolicy rp : policies) {
+            if (resourcePolicyService.isDateValid(rp)) {
+                return false;
+            }
```

Comment:
This seems odd to me, but maybe I'm misunderstanding. It looks like you are looping through READ policies to see if *any* have valid dates (or no dates attached at all). If they are valid, you assume it's not restricted?  What about a READ policy with a valid date which limits to a specific, non-Anonymous group?  In that scenario, it seems your `isRestricted()` would return false, when the object is restricted to a specific group.

Maybe I'm misunderstanding the logic here though... or maybe this method should be renamed so it doesn't imply that it's checking whether the object is access restricted (e.g. could be called `hasValidReadPolicies()`)

Reply:
The logic here is copied from https://github.com/DSpace/DSpace/blob/dspace-5_x/dspace-jspui/src/main/java/org/dspace/app/webui/util/RequestItemManager.java and I didn't like it either.

But consider:  the point of an item request is to bypass the normal DSpace authorization and let a human make the decision.  The intent of the code which calls this method (which I also copied from JSPUI logically if not quite verbatim) seems to be that if all bitstreams of an item are requested, a bitstream is not of an internal type (like licenses) and the bitstream is restricted *for anyone* (not necessarily everyone) then it should be attached to the email.  IOW we don't send copies of things that an anonymous user could get for himself.

I think that XMLUI did not make this distinction -- it exists there but is commented out.  Maybe we should just drop the distinction, and the method which embodies it.

Reply:
I'm honestly still a bit confused by it.  That said, it sounds like it serves a purpose & aligns with JSPUI behavior...so, I'm not against leaving it here, it just seems very oddly named. It's obviously not checking if something "isRestricted".  It seems to be checking if an object has  (any) READ policies that are all still "valid" (unexpired), which is a completely different concept than checking whether the object is access restricted.